### PR TITLE
RCNN upgrade to support TensorRT 8.

### PR DESCRIPTION
**Pull request by nengwp**
rcnn upgrade to support TensorRT 8.

Tested, on docker from: hakuyyf/tensorrtx:trt8.2_cuda11.2

### CODE REVIEW

File: rcnn/BatchedNms.cu [hunk 1]
```diff
@@ -63,7 +63,7 @@ int batchedNms(int batch_size,
         workspace_size += get_size_aligned<float>(count);  // scores_sorted
 
         size_t temp_size_sort = 0;
-        thrust::cuda_cub::cub::DeviceRadixSort::SortPairsDescending(
```
Comment (wang-xinyu) [inline rcnn/BatchedNms.cu:66]: Why do we need to remove this namespace `thrust::cuda_cub::`?

**Comment by wang-xinyu** [comment]
@nengwp Can we make it compatible with both TRT 7 and TRT 8? Like what we did in [yolov5/plugin/yololayer.](https://github.com/wang-xinyu/tensorrtx/blob/master/yolov5/plugin/yololayer.h)

File: rcnn/RpnDecodePlugin.h [hunk 2]
```diff
@@ -37,6 +37,7 @@ class RpnDecodePlugin : public IPluginV2Ext {
     mutable int size = -1;
 
  protected:
+    // 反序列化
```
Comment (wang-xinyu) [inline rcnn/RpnDecodePlugin.h:40]: 不要在代码里加任何中文呀

Comment (nengwp) [reply rcnn/RpnDecodePlugin.h:40]: 我将重新检查并删除这些中文

File: rcnn/BatchedNms.cu (hunk 1 above)
Comment (nengwp) [reply rcnn/BatchedNms.cu:66]: This should be an API interface change, where
#include <thrust/system/cuda/detail/cub/device/device_radix_sort.cuh>
change to
#include <cub/device/device_radix_sort.cuh>
Therefore, its namespace has changed, and the original path can no longer find the file

**Comment by nengwp** [comment]
> @nengwp Can we make it compatible with both TRT 7 and TRT 8? Like what we did in [yolov5/plugin/yololayer.](https://github.com/wang-xinyu/tensorrtx/blob/master/yolov5/plugin/yololayer.h)

I did not test it in trt7, and I think the code may not be supported for trt8 due to API changes. I have no current plan to modify their compatibility. Maybe add subfolders to support trt8, or wait for me to make compatibility improvements later.

**Comment by nengwp** [comment]
I'll close the commit and maybe reopen it when I test for compatibility

**Pull request closed without merging**